packages/test/test-drivers/src/odspTestDriver.ts

```diff
@@ -191,13 +196,18 @@ export class OdspTestDriver implements ITestDriver {
 				? Math.random()
 				: OdspTestDriver.legacyDriverUserRandomIndex;
 		const userIndex = Math.floor(randomUserIndex * creds.length);
-		const { username, password } = creds[userIndex];
+		// Bounds check above guarantees non-null (at least at compile time, assuming all types are respected)
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		const { username, password } = creds[userIndex]!;
 
 		const emailServer = username.substr(username.indexOf("@") + 1);
 
 		let siteUrl: string;
 		let tenantName: string;
-		if (emailServer.startsWith("http://") || emailServer.startsWith("https://")) {
+		if (
+			emailServer.startsWith("http://") === true ||
+			emailServer.startsWith("https://") === true
+		) {
```

Comment:
Interesting, the linter complained here? `startsWith()` should already be typed as boolean so I wouldn't expect a need to be explicit here...

Reply:
weird I thought it did, when I reverted it, it didnt complain. I have reverted this change